State state
---
Action next_action
